docs/shared/react-standalone-tutorial/4-workspace-optimization.md

```diff
@@ -0,0 +1,255 @@
+✓ 43 modules transformed.
+dist/store/index.html                  0.46 KiB
+dist/store/assets/index.50de2671.css   0.03 KiB / gzip: 0.05 KiB
+dist/store/assets/index.f18c2b19.js    157.69 KiB / gzip: 51.26 KiB
+dist/store/assets/index.f18c2b19.js.map 565.79 KiB
+
+>  NX  Vite build finished ...
+
+
+>  NX  Vite files available in dist/store
+
+
+ ————————————————————————————————————————————————————————————————————————————————————————————————————————
+
+ >  NX   Successfully ran target build for project store and 2 task(s) it depends on (6s)
+
+```
+
+Since you have not run the `build` target before for the `store` project, Nx runs the `build`, and populates the results in `dist/store` as specified in the `store` project's `project.json` file for the `build` target.
+
+Next, remove your dist directory:
+
+```shell
+rm -rf dist
+```
+
+And run the command again:
+
+```{% command="npx nx build admin" path="~/myorg" %}
+
+   ✔    2/2 dependent project tasks succeeded [2 read from cache]
+
+   Hint: you can run the command with --verbose to see the full dependent project outputs
+
+ ————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————
+
+
+> nx run store:build:production  [local cache]
+
+
+>  NX  Vite build starting ...
+
+vite v3.2.4 building for production...
+✓ 43 modules transformed.
+dist/store/index.html                  0.46 KiB
+dist/store/assets/index.50de2671.css   0.03 KiB / gzip: 0.05 KiB
+dist/store/assets/index.f18c2b19.js    157.69 KiB / gzip: 51.26 KiB
+dist/store/assets/index.f18c2b19.js.map 565.79 KiB
+
+>  NX  Vite build finished ...
+
+
+>  NX  Vite files available in dist/store
+
+
+ ————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————
+
+ >  NX   Successfully ran target build for project store and 2 task(s) it depends on (42ms)
+
+   Nx read the output from the cache instead of running the command for 3 out of 3 tasks.
+```
+
+Notice that `[local cache]` is mentioned in the terminal output, and that this time the command only took 42ms to run.
+
+Also notice that the result of your build has been added back to the `dist/store` directory.
+
+{% card title="More Task Caching Details" description="See the documentation for more information on caching." url="/core-features/cache-task-results" /%}
+
+## Configuring Task Pipelines
+
+Notice this line:
+
+```text
+   ✔    2/2 dependent project tasks succeeded [2 read from cache]
+```
+
+This is because your `store` project depends on the `cart` and `shared-ui` projects, which also have `build` targets. By default Nx is configured to run the `build` target for any dependencies that have a `build` target, before running the `build` on the original project.
+
+This feature allows the Nx graph to dynamically maintain task dependencies, rather than having to manually maintain those task dependencies as your workspace continues to grow.
```

Comment:
I think if we talk about what the feature does and show the config, this sentence can be removed.